lib/components/app/app-menu.tsx

```diff
@@ -163,17 +184,55 @@ class AppMenu extends Component<
 
   render() {
     const {
+      activeLocale,
       callTakerEnabled,
+      configLanguages,
+      createOrUpdateUser,
       extraMenuItems,
       fieldTripEnabled,
       intl,
+      loggedInUser,
       mailablesEnabled,
       popupTarget,
       resetAndToggleCallHistory,
       resetAndToggleFieldTrips,
+      setLocale,
       toggleMailables
     } = this.props
 
+    const languageMenuItems: menuItem[] | undefined = configLanguages && [
+      {
+        children: Object.keys(configLanguages)
+          .filter((locale) => locale !== 'allLanguages')
+          .map((locale) => ({
+            iconType: <></>,
+            id: configLanguages[locale].name,
+            label:
+              activeLocale === locale ? (
+                <strong>{configLanguages[locale].name}</strong>
+              ) : (
+                configLanguages[locale].name
+              ),
+            onClick: (e: MouseEvent) =>
+              handleLocaleSelection(
+                // @ts-expect-error TODO: correct this type pipeline
+                e,
+                locale,
+                activeLocale,
+                loggedInUser,
+                createOrUpdateUser,
+                setLocale,
+                intl
+              ),
+            subMenuDivider: false
+          })),
+        iconType: <GlobeAmericas />,
+        id: 'app-menu-locale-selector',
+        label: <FormattedMessage id="components.SubNav.languageSelector" />,
```

Comment:
Tricky a11y: To avoid setting up radio button elements or roles in the submenu items, maybe add a mention about the current language here so the label reads, with a screen reader: "Select language - current language: English". ("Current language..." can be visually hidden)